src/psyclone/core/signature.py

```diff
@@ -75,12 +75,26 @@ def __init__(self, variable, sub_sig=None):
             self._signature = (str(variable),) + sub_tuple
         elif isinstance(variable, tuple):
             self._signature = variable + sub_tuple
+        elif isinstance(variable, list):
+            self._signature = tuple(variable) + sub_tuple
         elif isinstance(variable, Signature):
             self._signature = variable._signature + sub_tuple
         else:
             raise InternalError("Got unexpected type '{0}' in Signature "
                                 "constructor".format(type(variable).__name__))
 
+    # ------------------------------------------------------------------------
+    @property
+    def is_structure(self):
```

Comment:
The Signature has been designed to kind of 'cache' (or give easy access to) information needed in the rest of the variable usage handling code. So yes, we could use something like:
```
if isinstance(var_access[0].node, StructureReference)
```
but:
1. I don't like using `isinstance` - it hard-codes the name of the class `StructureReference`, which is not necessary (which could be avoided by adding an `is_structure` method to the symbol, similar to the method for checking if a variable is an array).
2. We might not always have a reference (that might be due to a bug, but afaik at this stage the loop variable in a loop statement does not have a reference).
3. This information is fully derived from the PSyIR structure of the access, so it will seamlessly work if anything there will be changed.
4. This code is much easier than actually getting access to the var_access structure. Actually, looking at the code in question: there **is** no Reference I could even use: 
```
        all_props = api_config.grid_properties
...
        for name, prop in all_props.items():
            new_sig = Signature(prop.fortran.split("%"))
            if new_sig.is_structure:
```
This loop is over the grid properties of the config file. I don't like the `split("%")` statement (and would prefer if during parsing of the config file a psyir structure for this is created, but the properties are using "{0}" to insert the field name, so it's not that easy to get this into PSyIR to create a Signature). But ... there is no Reference  I could use. Technically, I could search for a "%" in the string, but I really prefer this approach, since it will make it easier if in some distant future we might support different input languages that do not use "%". And I think it better documents what the IF test is meant for.

So, tbh I don't see any problem using the Signature for exactly the job it was designed, simplifying the access information handling :grin:

Reply:
As usual, I'm forced to agree :-) One day, I'll be right (he hopes).